lib/src/jsonext_base.dart

```diff
@@ -29,14 +29,23 @@ extension JsonParse on Json {
   /// is null, return null.
   String? asStringN(String key) => this[key]?.toString();
 
-  /// Try to get value at [key] as bool. If the key does not exist or value is
-  /// null, return [fallback].
+  /// Try to get value at [key] as bool. If the key does not exist or value
+  /// can't be converted to bool, return [fallback].
   bool asBool(String key, {bool fallback = false}) =>
-      this[key] as bool? ?? fallback;
+      this.asBoolN(key) ?? fallback;
 
   /// Try to get value at [key] as bool. If the key does not exist or value
-  /// is null, return null.
-  bool? asBoolN(String key) => this[key] as bool?;
+  /// can't be converted to bool, return null.
+  bool? asBoolN(String key) {
+    if (this[key] is bool) return this[key] as bool;
+
+    final str = this[key]?.toString().toLowerCase();
+    return str == 'true'
```

Comment:
This is not required either. If we support this, then we will also need to think about 'True', 'False', etc. Let's keep it simple and support strictly boolean values.

Reply:
Never mind, I think this is fine. Thank you for your contribution.